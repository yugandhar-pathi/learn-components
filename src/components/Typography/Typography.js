import PropTypes from 'prop-types'

export const Typography = ({dataTestId, variant1, variant2, variant3, variant4, variant5, variant6, text}) =>{
    let val;
    if(variant1){
      val =  <h1>{text}</h1>
    }
    if(variant2){
        val =  <h2>{text}</h2>
      }
      if(variant3){
        val =  <h3>{text}</h3>
      }
      if(variant4){
        val =  <h4>{text}</h4>
      }
      if(variant5){
        val =  <h5>{text}</h5>
      }
      if(variant6){
        val =  <h6>{text}</h6>
      }
    return(
        <p data-testid={dataTestId}>{val}</p>
    )
}

Typography.propTypes = {
variant1: PropTypes.string,
variant2: PropTypes.string,
variant3: PropTypes.string,
variant4: PropTypes.string,
variant5: PropTypes.string,
variant6: PropTypes.string,
Text: PropTypes.string,
dataTestId: PropTypes.string.isRequired
}