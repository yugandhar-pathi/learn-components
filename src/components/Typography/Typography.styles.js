import styled from 'styled-components'
export const TypographyStyles = styled.p`


`