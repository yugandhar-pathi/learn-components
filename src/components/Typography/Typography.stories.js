import React from 'react';
import { Typography } from './Typography';

export default {
    component : "Typography",
}

export const variant1 = () => <Typography text="Hello World" variant1/>
export const variant2 = () => <Typography text="Hello World" variant2/>
export const variant3 = () => <Typography text="Hello World" variant3/>
export const variant4 = () => <Typography text="Hello World" variant3/>
export const variant5 = () => <Typography text="Hello World" variant4/>
export const variant6 = () => <Typography text="Hello World" variant5/>
