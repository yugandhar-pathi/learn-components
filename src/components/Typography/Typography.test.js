import {render, screen} from '@testing-library/react'
import { Typography } from './Typography'

test("Typography component",()=>{
    render(<Typography dataTestId= "typography-id"variant1="one" variant2="two" variant3="three" variant4="four" variant5="five" variant6="six" text="my text"/>)
    expect(screen.getByTestId("typography-id").textContent).toBe("my text")
})

test("snapshot for Typography component",()=>{
    render(<Typography dataTestId= "typography-id"variant1="one" variant2="two" variant3="three" variant4="four" variant5="five" variant6="six" text="my text"/>)
    expect(screen.getByTestId("typography-id")).toMatchSnapshot()
})