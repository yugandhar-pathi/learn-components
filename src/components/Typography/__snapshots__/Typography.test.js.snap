// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for Typography component 1`] = `
<p
  data-testid="typography-id"
>
  <h6>
    my text
  </h6>
</p>
`;
